### PR TITLE
Dev Setup on macOS

### PR DESCRIPTION
**Pull request by jeffmaher**
Had some trouble getting setup using Docker Compose and the instructions in CONTRIBUTING.md (perhaps because I don't have access to the Shopify `dev` tool). Wrote some verbose instructions that got me up and running on macOS.

See also:
- https://github.com/CovidShield/server/issues/24

### CODE REVIEW

File: docs/Dev Setup macOS.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+1. Run the build: `make`
+
+This will place built files into `./build/release`.
+
+## Setup the Database
+
+1. Setup some dev environment variables: `source scripts/dev_env_vars.sh`
+1. Setup the database: `./build/release/key-retrieval migrate-db`
+
+This will leave up the Key Retrieval server running at `http://localhost:8000`. You should see something like:
```
Comment (LouisStAmour) [inline docs/Dev Setup macOS.md:83]: The key retrieval server is at `:8001` not `:8000`

**Comment by LouisStAmour** [comment]
There's also some preliminary documentation at https://github.com/CovidShield/server/blob/7530e2189088653f5e5dbf2c1b53e9731e0274b4/config/infrastructure/aws/README.md

File: docs/Dev Setup macOS.md (hunk 1 above)
Comment (jeffmaher) [reply docs/Dev Setup macOS.md:83]: Thanks for the find! Will update.

File: docs/Dev Setup macOS.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+1. Setup the following environment variables:
+    ```
+    export DB_USER=<username>
+    export DB_PASS=<password>
+    export DB_HOST=<hostname>
+    export DB_NAME=<test database name>
+    ```
+1. Install dependencies for the tests: `bundle install`
+1. Run the tests: `make test`
```
Comment (jeffmaher) [inline docs/Dev Setup macOS.md:130]: Could use some help from the core team on this last part, which doesn't work yet. I've confirmed that the database is setup and running, but that the tests still can't see it.

Here's an active connection to the database via a DB GUI tool (TablePlus): 
Screen Shot 2020-05-24 at 09 02 25

The database config is setup in `scripts/dev_env_vars.sh` (included in this PR):
```
export DB_USER=root
export DB_PASS=somepasswordhere
export DB_HOST=localhost
export DB_NAME=covidshield
```

Here's the log of running `make test`:
[test_log.txt](https://github.com/CovidShield/server/files/4673559/test_log.txt)

Comment (jeffmaher) [reply docs/Dev Setup macOS.md]: All fixed! Thanks for the help team!

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -80,3 +80,134 @@ Finally, run:
...
+
+### Install Go Lang
+
+The application has been tested with the Go programming language at version 1.14.x. This installs that version explicitly.
+
+`brew install go@1.14`
+
+### Install Ruby
+
+This sets up your Mac to run version 2.6.5 of Ruby, which is what the application has been tested at as of writing.
```
[the remark this answers is not in the document]
Comment (jeffmaher) [reply CONTRIBUTING.md]: Bumped 👍 . (`docker-compose` seems to be working on my end! Setting up the Portal reference app today using that, but if I find anything there, will open a new Issue or PR.)

**Comment by honkfestival** [comment]
@jeffmaher @LouisStAmour I'm happy to merge this if you'd like, but I don't think we need it if `docker-compose` works okay. What do you think?

**Comment by LouisStAmour** [comment]
I’m not speaking on behalf of any agency or organization here, this is just my thoughts as an individual developer:

I can’t say for sure this branch should be merged, but in the sprit of #24 I think it would be helpful to share instructions for getting started in your IDE that don’t involve an internal Shopify tool. If that means a brief set of steps on getting a local golang/ruby/MySQL environment set up via build and runtime containers — so you’re editing and compiling files in Docker — or if that means setting up local (global or rvm/gvm) versions of build toolchains, requirements, etc. It would be helpful for new contributors who don’t have access to internal Shopify tools.

That said, a related story is posting public milestones around what functionality is still to be added, and with external partners, establishing either a release process with expected configuration points at each release, or otherwise supporting regional “forks” by including documentation on what has changed between releases. I’m not saying this would happen right away, but the lack of clear guidance on adoption around a rapidly changing codebase with some stubbed out components is going to be a blocker either for initial adoption or for later changes after an agency puts this out in beta.

Apologies if this is too early to bring this up, but at some point the list of contributors will grow to include devs and ops from external groups and we should try to prepare for that somehow. Documentation and release notes, or getting folks up to speed on what’s in the repo now or soon, is incredibly useful for this! 👍

**Comment by jeffmaher** [comment]
> @jeffmaher @LouisStAmour I'm happy to merge this if you'd like, but I don't think we need it if `docker-compose` works okay. What do you think?

I'm OK ditching/closing this now that Docker Compose is working.

To @LouisStAmour 's comment, it might be worthwhile to update CONTRIBUTING.md to have the Docker Compose instructions. (For example, if someone is a developer that uses VS Code, they'll then know that it's safe to use Docker Compose with something like VS Code's Remote Container extension.)

**Pull request closed without merging**